lib/AST/Module.cpp

```diff
@@ -549,6 +549,11 @@ void ModuleDecl::lookupObjCMethods(
   FORWARD(lookupObjCMethods, (selector, results));
 }
 
+void ModuleDecl::lookupImportedSPIs(const ModuleDecl *importedModule,
```

Comment:
Is the module variant of this called anywhere? Generally, imports in other source files don't affect behavior in a source file. So the decl checker should always be calling this on a SourceFile and not the entire module. Also, FORWARD is O(n) in the number of source files -- this can be slow if you're calling it a lot.

Reply:
`lookupImportedSPIs` is called on modules when printing the textual Swift interface and for serialization (including at merge-modules), so it's not called often. I'm not sure we could keep in only on the source file in this case.

However, the SPI decl checks use `isImportedAsSPI` which is only on source files and it calls only `SourceFile:: lookupImportedSPIs`.